File: 0205-isomorphic-strings/0205-isomorphic-strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        vector<int> snums(256,0);
        vector<bool> tnums(256,false);
        for(int i=0;i<=s.size();i++){
            if(!snums[s[i]] && !tnums[t[i]]){
                snums[s[i]]=t[i];
                tnums[t[i]]=true;
            }
        }
        for(int i=0;i<=s.size()-1;i++){
            if(snums[s[i]]!=t[i]) return false;
        }
        return true;
    }
};

File: 0205-isomorphic-strings/0205-isomorphic-strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        if(s.length()!=t.length()) return false;
        char[] ch1=new char[128];
        char[] ch2=new char[128];
        for(int i=0;i<s.length();i++)
        {
            int j=(int)s.charAt(i);
            if(ch1[j]=='\0') 
            ch1[j]=t.charAt(i);
            else
            if(ch1[j]!=t.charAt(i)) return false;
        }
        for(int i=0;i<s.length();i++)
        {
            int j=(int)t.charAt(i);
            if(ch2[j]=='\0') 
            ch2[j]=s.charAt(i);
            else
            if(ch2[j]!=s.charAt(i)) return false;
        }
        return true;
    }
}